Initial SPM Package

devahmedshendy opened the pull request.

devahmedshendy committed:
package init

devahmedshendy committed:
Add iOS 16.0 platform support

devahmedshendy committed:
Add playground project

devahmedshendy committed:
Implement MaterialText

0xLeif commented on Package.swift:
```diff
@@ -0,0 +1,34 @@
+let package = Package(
+    name: "OpenMaterialUI",
+    platforms: [
+        .iOS(.v16)
```
Should we add mac, watch, and tv?

0xLeif commented on Sources/OpenMaterialUI/Text/MaterialText.swift:
```diff
@@ -0,0 +1,40 @@
+
+import SwiftUI
+
+@available(iOS 16.0, *)
```
We shouldn't need this if the Package.swift file specifies we are using iOS 16.

0xLeif commented on playground/OpenMaterialUIPlayground/ContentView.swift:
```diff
@@ -0,0 +1,27 @@
+            Image(systemName: "globe")
+                .imageScale(.large)
+                .foregroundColor(.accentColor)
+            Text("Hello, world!")
```
Can we change this to use `MaterialText`?

0xLeif commented on playground/OpenMaterialUIPlayground.xcodeproj/project.pbxproj:
Nitpick: Can we update the `playground` folder to `Playground`? Normally folders start with a capital letter.

devahmedshendy replied:
@0xLeif  I will have this committed in another PR.

devahmedshendy committed:
Update platforms, remove @available annotation:
- update platforms to include macOs, tvOS, watchOS.
- use MaterialText in playground sample
- remove @available annotation

devahmedshendy committed:
update: clean Package.swift

devahmedshendy committed:
update: cleanup and add MaterialTextDemo.swift
